Quaver.Shared/IPC/QuaverIpcHandler.cs

```diff
@@ -54,7 +62,78 @@ public static void HandleProtocolMessage(string message)
                 if (game?.CurrentScreen is EditScreen screen)
                     screen.GoToObjects(message);
                 else
-                    NotificationManager.Show(NotificationLevel.Warning, "You must be in the editor to use this function!");
+                    NotificationManager.Show(NotificationLevel.Warning,
+                        "You must be in the editor to use this function!");
+            }
+            else if (message.StartsWith("map/"))
+            {
+                var game = (QuaverGame) GameBase.Game;
+                if (game.CurrentScreen.Type != QuaverScreenType.Select)
+                {
+                    NotificationManager.Show(NotificationLevel.Warning,
+                        "You must be in the song select screen to select a map!");
+                    return;
+                }
+
+                var screen = game.CurrentScreen as SelectionScreen;
+
+                message = message.Replace("map/", "");
+                int mapId;
+                if (!int.TryParse(message, out mapId))
+                {
+                    NotificationManager.Show(NotificationLevel.Warning, "The provided ID was not a number!");
```

Comment:
I'd do parsing checks before the screen check I think so people don't have to quit what they're doing just to find out the message was malformed